webpack/assets/javascripts/react_app/components/HostsIndex/BulkActions/helpers.js

```diff
@@ -30,3 +33,35 @@ export const failedHostsToastParams = ({
 
   return toastParams;
 };
+
+export const useHostsBulkSelect = ({ initialSelectedHosts, modalIsOpen }) => {
+  const defaultParams = { search: initialSelectedHosts };
+  const apiOptions = { key: 'HOST_REVIEW' };
+  const replacementResponse = !modalIsOpen ? { response: {} } : false;
+  const hostsResponse = useTableIndexAPIResponse({
+    replacementResponse, // don't fetch data if modal is closed
+    apiUrl: `${HOSTS_API_PATH}?per_page=7`,
```

Comment:
Why a magic `7` here?
```suggestion
    apiUrl: HOSTS_API_PATH,
```

Or did you intend `per_page=all` to avoid pagination? If so, I think that should be passed via `defaultParams` and not as part of the URL.

Reply:
I'll take responsibility for this, as I assume it was copied and pasted from one of the wizards on the same page. A per_page = 7 was the starting page size that looks the best with these modals. It doesn't make sense to start with the default per_page of 20 in this case, because they don't all fit on the screen.

Reply:
It may be. On the non-react all host page you can also select all items. Also, the user may have changed the page size on the overview to be more than the default. So I'd suggest `per_page=all` to avoid pagination.